app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :ensure_superuser
+
+  # GET /users
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  def show
+  end
+
+  # GET /users/1/edit
+  def edit
+  end
+
+  # PATCH/PUT /users/1
+  def update
+    if params[:toggle]
+      if @user.has_role? :admin
+        @user.remove_role :admin
+      else
+        @user.add_role :admin
+      end
+    end
+    redirect_to users_path
+  end
+
+  # DELETE /users/1
+  def destroy
+    @user.destroy
+    redirect_to users_url, notice: 'User was successfully destroyed.'
```

Comment:
Instead of passing notice like this for deletion, it is more common to just show popup message.

Reply:
Changed.
I still use notice, but now I changed in the views to display the notice as a pop up message